WordPress/Classes/Services/PostService.swift

```diff
@@ -0,0 +1,599 @@
+import Foundation
+
+class PostService : LocalCoreDataService {
+    static let ErrorDomain = "PostServiceErrorDomain"
+    static let DefaultNumberToSync: UInt = 40
+
+    enum PostType: String {
+        case any = "any"
+        case post = "post"
+        case page = "page"
+
+        func type() -> AbstractPost.Type {
+            switch self {
+            case .post: return Post.self
+            case .page: return Page.self
+            case .any: return AbstractPost.self
+            }
+        }
+    }
+
+    typealias SuccessPosts = ([AbstractPost]) -> Void
+    typealias FailureBasic = (NSError?) -> Void
+    typealias SuccessBasic = () -> Void
```

Comment:
I think we might avoid having a "basic" completion type in this case, and instead just default to a `() -> Void` where needed.